WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -27,105 +21,67 @@ import { groupTypes, delegateRegionsStatus } from '../../lib/wca-data.js.erb';
 // i18n-tasks-use t('enums.user.role_status.councils.senior_member')
 // i18n-tasks-use t('enums.user.role_status.councils.leader')
 
+const sortParams = 'status';
+
 export default function RolesTab({ userId }) {
-  const roleListFetch = useLoadedData(rolesOfUser(
+  const {
+    data: activeRoles,
+    loading: activeRolesLoading,
+    error: activeRolesError,
+    sync: activeRolesSync,
+  } = useLoadedData(apiV0Urls.userRoles.listOfUser(
     userId,
+    sortParams,
     { isActive: true, isGroupHidden: false },
   ));
-  const { loggedInUserPermissions, loading } = useLoggedInUserPermissions();
+  const {
+    data: pastRoles,
+    loading: pastRolesLoading,
+    error: pastRolesError,
+    sync: pastRolesSync,
+  } = useLoadedData(apiV0Urls.userRoles.listOfUser(
+    userId,
+    sortParams,
+    { isActive: false, isGroupHidden: false },
+  ));
 
   const [open, setOpen] = React.useState(false);
 
-  const isDelegate = roleListFetch.data && roleListFetch.data.some(
-    (role) => role.group.group_type === 'delegate_regions',
+  const isDelegate = activeRoles?.some(
+    (role) => role.group.group_type === groupTypes.delegate_regions,
   );
 
-  function hyperlink(role) {
-    if (role.group.group_type === groupTypes.delegate_regions) {
-      if ([
-        delegateRegionsStatus.senior_delegate,
-        delegateRegionsStatus.regional_delegate,
-      ].includes(role.metadata.status)) {
-        return panelUrls.board.regionsManager;
-      }
-      return null;
-    }
-    if (role.group.group_type === groupTypes.teams_committees) {
-      return `${teamUrl(role.group.id.split('_').pop())}/edit`;
-    }
-    if (role.group.group_type === groupTypes.translators) {
-      return panelUrls.wst.translators;
-    }
-    return null;
-  }
-
-  function isHyperlinkableRole(role) {
-    if (role.group.group_type === groupTypes.delegate_regions) {
-      return [
-        delegateRegionsStatus.senior_delegate,
-        delegateRegionsStatus.regional_delegate,
-      ].includes(role.metadata.status);
-    }
-    return [groupTypes.teams_committees, groupTypes.translators].includes(role.group.group_type);
-  }
-
-  function getRoleDescription(role) {
-    let roleDescription = '';
-    if (role.metadata.status) {
-      roleDescription += `${I18n.t(`enums.user.role_status.${role.group.group_type}.${role.metadata.status}`)}, `;
-    }
-    roleDescription += role.group.name;
-    return roleDescription;
+  function hasAtLeastOneRole() {
+    return activeRoles?.length > 0 || pastRoles?.length > 0;
   }
 
-  if (roleListFetch.loading || loading) return <Loading />;
-  if (roleListFetch.error) return <Errored />;
+  if (activeRolesLoading || pastRolesLoading) return <Loading />;
+  if (activeRolesError || pastRolesError) return <Errored />;
 
   return (
     <>
-      {roleListFetch.data?.length > 0
+      {hasAtLeastOneRole()
         ? (
```

Comment:
This ternary might be cleaner as
```
<>
  {activeRoles?.length > 0 && (
    ...
  )}
  {pastRoles?.length > 0 && (
    ...
  )}
  {hasNoRoles && (
    ...
  )}
</>
```
Note the use of `hasNoRoles` rather than `hasAtLeastOneRole`.

Reply:
Looks good, done.